pkg/controller/command/verifiable/command.go

```diff
@@ -303,32 +385,207 @@ func (o *Command) GeneratePresentationByID(rw io.Writer, req io.Reader) command.
+	// prepare vp
+	vp, err := o.createAndSignPresentation(vcs, pk)
 	if err != nil {
-		logutil.LogError(logger, commandName, commandMethodName, "generate vp : "+err.Error())
+		logutil.LogError(logger, commandName, generatePresentationCommandMethod, "prepare vp: "+err.Error())
 
-		return command.NewValidationError(errCode, fmt.Errorf("generate vp : %w", err))
+		return command.NewValidationError(GeneratePresentationByIDErrorCode, fmt.Errorf("prepare vp: %w", err))
 	}
 
-	// TODO https://github.com/hyperledger/aries-framework-go/issues/1422 - add proof
+	return o.generate(rw, vp)
+}
 
+func (o *Command) generatePresentationByID(rw io.Writer, vc *verifiable.Credential, didDoc *did.Doc) command.Error {
+	// prepare vp by id
+	vp, err := o.createAndSignPresentationByID(vc, didDoc)
+	if err != nil {
+		logutil.LogError(logger, commandName, generatePresentationCommandMethod, "prepare vp by id: "+err.Error())
+
+		return command.NewValidationError(GeneratePresentationByIDErrorCode, fmt.Errorf("prepare vp by id: %w", err))
+	}
+
+	return o.generate(rw, vp)
+}
+
+func (o *Command) generate(rw io.Writer, vp *verifiable.Presentation) command.Error { // nolint: interfacer
 	vpBytes, err := vp.MarshalJSON()
 	if err != nil {
-		logutil.LogError(logger, commandName, commandMethodName, "generate vp : "+err.Error())
+		logutil.LogError(logger, commandName, generatePresentationCommandMethod, "generate vp: "+err.Error())
 
-		return command.NewValidationError(errCode, fmt.Errorf("generate vp : %w", err))
+		return command.NewValidationError(GeneratePresentationByIDErrorCode, fmt.Errorf("generate vp : %w", err))
 	}
 
 	command.WriteNillableResponse(rw, &Presentation{
 		VerifiablePresentation: string(vpBytes),
 	}, logger)
 
-	logutil.LogDebug(logger, commandName, commandMethodName, "success")
+	logutil.LogDebug(logger, commandName, generatePresentationCommandMethod, "success")
 
 	return nil
 }
+
+func (o *Command) createAndSignPresentation(credentials []*verifiable.Credential,
+	pk string) (*verifiable.Presentation, error) {
+	var err error
+
+	var vp *verifiable.Presentation
+
+	var vcs []interface{}
+	// loop through all credentials present in the request and create unsigned presentation
+	for i, vc := range credentials {
+		if i == 0 {
+			vp, err = vc.Presentation()
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		vcs = append(vcs, vc)
+	}
+	// Add array of credentials in the presentation
+	err = vp.SetCredentials(vcs...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set credentials: %w", err)
+	}
+
+	// Add proofs to vp - sign presentation
+	vp, err = o.addLinkedDataProof(vp, pk)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign vp: %w", err)
+	}
+
+	return vp, nil
+}
+
+func (o *Command) createAndSignPresentationByID(vc *verifiable.Credential,
+	didDoc *did.Doc) (*verifiable.Presentation, error) {
+	// pk is verification method
+	pk, err := getDefaultVerificationMethod(didDoc)
```

Comment:
Shouldnt we check for options before picking up default verification method?

Reply:
so that means i should add proof options in presentaRequestByID model. I only added proof option in generatePresentation (input vc) . Shall i add for both or it can be part of #1644

 // PresentationRequestByID model
//
// This is used for querying/removing by ID from input json.
//
type PresentationRequestByID struct {
	// ID
	ID string `json:"id"`

	// DID ID
	DID string `json:"did"`
}